plugins/auth-backend/src/lib/resolvers/CatalogAuthResolverContext.ts

```diff
@@ -139,19 +139,59 @@ export class CatalogAuthResolverContext implements AuthResolverContext {
     return { entity: result };
   }
 
-  async signInWithCatalogUser(query: AuthResolverCatalogUserQuery) {
-    const { entity } = await this.findCatalogUser(query);
+  async signInWithCatalogUser(
+    query: AuthResolverCatalogUserQuery,
+    options?: {
+      dangerousEntityRefFallback?:
```

Comment:
Suggestion:

```ts
dangerouslyAllowFallbackForMissingEntity?:
  | {
      entityRef: string | {
        kind?: string;
        namespace?: string;
        name: string;
      }
    }
  | undefined;
```

That makes it "a whole fallback" as a concept, and the one required field in it (for now) is the entityRef.